src/transaction.cpp

```diff
@@ -1086,12 +1096,12 @@ int wsrep::transaction::streaming_step(wsrep::unique_lock<wsrep::mutex>& lock)
         break;
     }
 
-    if (streaming_context_.fragment_size_exceeded())
+    if (streaming_context_.fragment_size_exceeded() || client_service_.is_xa_prepare())
     {
         // Some statements have no effect. Do not atttempt to
         // replicate a fragment if no data has been generated
-        // since last fragment replication.
-        if (bytes_to_replicate <= 0)
+        // since last fragment replication, and statement is not XA PREPARE.
```

Comment:
This comment is not very useful. Briefly explain that XA PREPARE may add some data down this path